netbox/netbox/views/generic/bulk_views.py

```diff
@@ -689,14 +689,17 @@ def _rename_objects(self, form, selected_objects):
 
             find = form.cleaned_data['find']
             replace = form.cleaned_data['replace']
-            if form.cleaned_data['use_regex']:
+            if obj.name and form.cleaned_data['use_regex']:
                 try:
-                    obj.new_name = re.sub(find, replace, obj.name or '')
+                    obj.new_name = re.sub(find, replace, obj.name)
                 # Catch regex group reference errors
                 except re.error:
                     obj.new_name = obj.name
-            else:
+            elif obj.name:
                 obj.new_name = obj.name.replace(find, replace)
+            else:
+                obj.new_name = obj.name
```

Comment:
This effectively bypasses the rename operation, which may be sufficient, however #14044 seems to expect that unnamed devices would be renamed.